app/[lang]/dictionaries.ts

```diff
@@ -0,0 +1,50 @@
+const dictionaries: LangDictionary = {
+  en: () => import('./dictionaries/en.json').then((module) => module.default),
+  fr: () => import('./dictionaries/fr.json').then((module) => module.default),
+  kw: () => import('./dictionaries/kw.json').then((module) => module.default),
+};
+
+export const getDictionary = async (lang: Lang) => {
+  const dictionaryFunction = dictionaries[lang];
+  if (typeof dictionaryFunction === 'function') {
+    return await dictionaryFunction();
+  }
+  return {}; // Return an empty object or handle the error as needed.
+};
+
+export const getFromDictionary = (
+  keys: Array<string>,
+  dict: Dict | string
+): Dict | string => {
+  if (typeof dict == 'string') {
+    return dict;
+  }
+
+  if (keys.length === 0) {
+    return '';
+  }
+
+  if (!dict) {
+    return '';
+  }
+  const key = keys.shift() || '';
+  return getFromDictionary(keys, dict[key]);
+};
+
+export const t = (key: string, dict: Dict): string => {
```

Comment:
I am not fully convinced it's cleaner to use this function + a dictonary to get localized strings instead of just the dictionary like they recommend in the NextJS-guide. Why not just the dictionary?

Reply:
If we are going to use the approach of directly accessing the dictionary, we have to define a new type which contains all the different levels of the i18n string hierarchy. We can discuss it later, but essentially it is less code using the translate function. It is also more similar to the normal approach with i18n used in other projects, as far as I have experienced